components/suggest/src/db.rs

```diff
@@ -126,6 +126,7 @@ impl<'a> SuggestDao<'a> {
     /// Fetches suggestions that match the given query from the database.
     pub fn fetch_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
         let (keyword_prefix, keyword_suffix) = split_keyword(&query.keyword);
+        let suggestions_limit = if let Some(l) = query.limit { l } else { -1 };
```

Comment:
```suggestion
        let suggestions_limit = query.limit.unwrap_or(-1);
```

`Option` has [a handful of convenience methods](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or) for unwrapping values; we can use one of those here! ✨

Reply:
This is so much nicer! I'll add this change, thanks! :)